src/Standards/Generic/Sniffs/PHP/ForbiddenFunctionsSniff.php

```diff
@@ -74,15 +74,15 @@ public function register()
 
         // If we are not pattern matching, we need to work out what
         // tokens to listen for.
-        $string = '<?php ';
+        $tokens = [];
         foreach ($this->forbiddenFunctionNames as $name) {
-            $string .= $name.'();';
+            $token = token_get_all(sprintf('<?php %s();', $name));
+            array_shift($token);
+            $tokens = array_merge($tokens, $token);
```

Comment:
Using `array_merge()` within a loop is highly inefficient.

Looking at your report in #2376, why not just look at the array key/value and skip (`continue`) if it's `__halt_compiler` ? and if found, add it to the end of the string before the tokenization ?

Reply:
Yes that would be a valid fix as well, and would have much better performance if this `Sniff` is instantiated many times.
My main reason for using this rather inefficient method is to avoid considering any edge/special cases (just in case some other functions would break `token_get_all` in the future).
If performance is an important concern here, I'll change it accordingly.

Reply:
I don't know of any other PHP function (nor of RFCs to add any) which would halt the compiler and therefore the tokenizer, so IMO it should be safe to change this.